src/storage/storage.ts

```diff
@@ -0,0 +1,809 @@
+    for await (const token of this.store.registeredTokenIter()) {
+      yield token;
+    }
+  }
+
+  /**
+   * Get a token from storage along with the metadata of the wallet transactions.
+   *
+   * @param {string|undefined} [token='00'] Token uid to fetch
+   * @returns {Promise<(ITokenData & Partial<ITokenMetadata>)|null>}
+   */
+  async getToken(token?: string): Promise<(ITokenData & Partial<ITokenMetadata>)|null> {
+    return this.store.getToken(token || HATHOR_TOKEN_CONFIG.uid);
+  }
+
+  /**
+   * Regsiter a token.
+   * @param {ITokenData} token Token data to register
+   * @returns {Promise<void>}
+   */
+  async registerToken(token: ITokenData): Promise<void> {
+    await this.store.registerToken(token);
+  }
+
+  /**
+   * Unregister a token from the wallet.
+   * @param {Promise<void>} tokenUid Token uid to unregister.
+   * @returns {Promise<void>}
+   */
+  async unregisterToken(tokenUid: string): Promise<void> {
+    await this.store.unregisterToken(tokenUid);
+  }
+
+  /**
+   * Iterate on all utxos of the wallet.
+   * @returns {AsyncGenerator<IUtxo, any, unknown>}
+   */
+  async *getAllUtxos(): AsyncGenerator<IUtxo, any, unknown> {
+    for await (const utxo of this.store.utxoIter()) {
+      yield utxo;
+    }
+  }
+
+  /**
+   * Select utxos matching the request and do not select any utxos marked for inputs.
+   *
+   * @param {Omit<IUtxoFilterOptions, 'reward_lock'>} [options={}] Options to filter utxos and stop when the target is found.
+   * @returns {AsyncGenerator<IUtxo, any, unknown>}
+   */
+  async *selectUtxos(options: Omit<IUtxoFilterOptions, 'reward_lock'> = {}): AsyncGenerator<IUtxo, any, unknown> {
+    const newFilter = (utxo: IUtxo): boolean => {
+      const utxoId = `${utxo.txId}:${utxo.index}`;
+      return (!this.utxosSelectedAsInput.has(utxoId)) && (options.filter_method ? options.filter_method(utxo) : true);
+    }
+
+    const newOptions: IUtxoFilterOptions = {
+      ...options,
+      filter_method: newFilter,
+    };
+    if (this.version?.reward_spend_min_blocks) {
+      newOptions.reward_lock = this.version.reward_spend_min_blocks;
+    }
+    for await (const utxo of this.store.selectUtxos(newOptions)) {
+      yield utxo;
+    }
+  }
+
+  /**
+   * Calculate the token balance of a transaction, including authorities.
+   * @param {IDataTx} tx The transaction we want to calculate the balance.
+   * @returns {Promise<Map<string, Record<'funds'|'mint'|'melt', number>>>} The balance of all tokens on the transaction.
+   */
+  async calculateTxBalance(tx: IDataTx): Promise<Map<string, Record<'funds'|'mint'|'melt', number>>> {
+    function getEmptyBalance(): Record<'funds'|'mint'|'melt', number> {
+      return {'funds': 0, 'mint': 0, 'melt': 0};
+    }
+    const balance = new Map<string, Record<'funds'|'mint'|'melt', number>>();
+
+    for (const output of tx.outputs) {
+      if (isDataOutputCreateToken(output)) {
```

Comment:
This method was created from `fillTx` during the review.
The `fillTx` method would require us to skip minted tokens since we cannot add inputs to match a token that is being created on the transaction.
I guess I can make this skip optional since it does not make sense as part of `calculateTxBalance`, what do you think?

Reply:
The whole method was removed since it was not necessary, create an utility for this which is more reflective of what it should do.